go-controller/pkg/ovn/policy.go

```diff
@@ -75,19 +77,40 @@ type networkPolicy struct {
 	portGroupName string
 	deleted       bool //deleted policy
 	created       bool
+
+	// only valid for policy that sharing portGroup, if sharePortGroupName is "", it cannot share portGroup
+	sharePortGroupName string
+	ingressDefDeny     bool
+	egressDefDeny      bool
+}
+
+// now we only support portGroup sharing if the policy's local pod selector is empty. This can be changed later.
+func getSharedPortGroupName(policy *knet.NetworkPolicy) string {
+	sel := &policy.Spec.PodSelector
+	if len(sel.MatchLabels) == 0 && len(sel.MatchExpressions) == 0 {
+		// now we only support empty podSelector policy to share the portGroup
+		return policy.Namespace + "_" + "Share_PortGrp"
```

Comment:
do we really need hte "Share_PortGrp", can't this just be  port group by namespace name?

Reply:
it is just to make it clear when looking at the logical entities, that this portgroup is shared by many policies.

Reply:
I don't know what is the plan for supporting shared port groups for non-empty selectors, but for empty selector we can just use [e/in]gressDefaulDeny port group based on policy type - it will contain all namespace pods?
Then we will have even less port groups and pod watchers, and the only change that should be required is delete network policy not with the whole port group, but only ACLs

Reply:
this is not true. ports of all policies of the given namespaces are added to the ingress/egressDefaultDeny port group as long as it is selected by the localPodSelector.

Also, like I said, the sharedPortGroup support can support not only empty selectors even now, with a single line change (that set its np.portGroupName). The issue is only 1. what selector we'd like to support if not all, 2. what group name we choose.

Reply:
Sorry, but I don't think i understand your explanation.
if the pod is no longer selected by the localPodSelector, then it will cause policy group change anyway (that will happen together with pod selector change)
So what I am saying is as long as network policy has empty selector it can use default deny port group. Correct me if that is wrong.
I also understand that this code will be needed to support other pod selectors, but I am still not sure about the use cases for multiple policies with the same non-empty pod selector - if you could explain why you expect users to create such policies it would help.

Reply:
@npinaeva selected local pods are not added to the default deny port group blindly, it is complicated and requires to carefully count the reference count by all the applied policies, see https://github.com/ovn-org/ovn-kubernetes/blob/dbf58734b9f1413ccb8f36d107cc11963aef141f/go-controller/pkg/ovn/policy.go#L731.